ui/src/pipeline/edit/transformation/CodeEditor.vue

```diff
@@ -18,16 +18,39 @@ import MonacoEditor, { MonacoEditorConstructor } from 'vue-monaco'
 import * as monaco from 'monaco-editor'
 
 import { JobResult, JobError } from './transformation'
+import { Action, State, Mutation } from 'vuex-class'
+
+const namespace = { namespace: 'transformation' }
 
 @Component({
   components: {
     MonacoEditor
   }
 })
 export default class CodeEditor extends Vue {
-  @Prop() readonly value!: string
-  @Prop() readonly data!: object
-  @Prop() readonly result!: JobResult | null
+  /** from vuex module */
+  @Action('data', namespace)
+  private data!: any
```

Comment:
object or unknown?

Reply:
well, i'm not sure. what kind of data can get returned from the adapter? Do we support APIs that only return a primitive (e.g. a number or a string?). Is an array of values covered by defining the type as `object` ?

Reply:
how about a union type like `object | object[]`?